R/text_to_date.R

```diff
@@ -0,0 +1,132 @@
+#' Extract dates from text
+#'
+#' Sometimes dates can be contained in text,
+#' this function extracts those dates from text and
+#' converts them to messydt objects.
+#' @param v A character vector
+#' @return Dates in messydt format
+#' @importFrom stringr str_replace_all str_squish str_extract word str_starts
+#' str_remove_all
+#' @details If multiple dates are identified in the same row,
+#' only the first date is returned.
+#' @examples
+#' v <- c("This function was created on the 29 September 2021",
+#' "Today is October 12, 2021", "Tomorrow is 13-10-2021")
+#' text_to_date(v)
+#' @export
+text_to_date <- function(v) {
```

Comment:
Why this function name?

1. the result will be a messydt class object, not date
2. `as_messydate.character()` already works on characters, maybe we just need to expand what formats it accepts?
3. if it's about extracting just the date-related strings, perhaps nesting an `extract_from_text()` or `isolate_dates()` within `as_messydt()` might string extract the date related aspects in a character string?

Reply:
I have added an extra "text" logical argument to `as_messydate()` that makes so the function accepts and converts dates within texts to messydates. Please let me know if this is what you mean here.